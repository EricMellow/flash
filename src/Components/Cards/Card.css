.Card {
  text-align: center;
  background-color: white;
  width: 40vw;
  box-shadow: 2px 2px 10px lightgrey;
  margin: auto;
  height: 225px
}

button {
  width: 100px;
  height: 40px;
  border-radius: 5px;
  margin: 10px;
  font-size: 1rem;
}

.yes-btn {
  background-color: #45b945;
}

.no-btn {
  background-color: #f14747;
}

.category {
  background-color: #7EB8CB;
  margin: 0;
  height: 30px;
  padding: 5px;
}

.question {
  padding: 5px;
}

