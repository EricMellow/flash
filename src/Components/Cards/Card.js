import React, { Component } from 'react';
import './Card.css';

class Card extends Component {
  constructor(props) {
    super()
    this.state = {
      showAnswer: false,
      correctAnswers: 0,
      timesSeen: 0
    }
  }

  showAnswer() {
    this.setState({
      showAnswer: !this.state.showAnswer
    })
  }
  
  render() {
    if (!this.state.showAnswer) {
      return (
        <div 
          className='Card' 
          aria-label="Flash card"
          onClick={() => {
          this.showAnswer()
        }}>
          <h2 className='category'>Category: {this.props.category}</h2>
          <p>You have answered this question correctly {this.state.correctAnswers} times out of {this.state.timesSeen}</p>
          <h4 className='question' aria-label="Question">{this.props.question}</h4>
          <p>*Click card to see the answer</p>
        </div>
      );
    } else {
      return (
        <div className='Card' aria-label="Flash card">
          <h2 className="category">Answer</h2>
          <h4 className="answer" aria-label="Answer">The correct answer is: {this.props.answer}</h4>
          <h4>Did you guess the correct answer?</h4>
          <button 
            className="yes-btn" 
            aria-label="Yes button"
            onClick={() => {
            let newScore = this.state.correctAnswers;
            let timesSeen = this.state.timesSeen
            this.setState({
              showAnswer: false,
              correctAnswers: newScore += 1,
              timesSeen: timesSeen += 1
            })
            this.props.updateScore(1)
          }}>Yes</button>
          <button 
            className="no-btn" 
            aria-label="No button"
            onClick={() => {
            let timesSeen = this.state.timesSeen
            this.setState({
              showAnswer: false,
              timesSeen: timesSeen += 1
            })
            this.props.updateScore(0)
          }}>No</button>
        </div>
      );
    }
  }
}
export default Card;
