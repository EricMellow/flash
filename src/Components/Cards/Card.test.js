import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { exportAllDeclaration } from '@babel/types';

Enzyme.configure({ adapter: new Adapter('enzyme-adapter-react-15') });

describe('Card', () => {
let wrapper;

beforeEach(() => {
  wrapper= shallow(<Card />);
})

it('should match the snapshot if showAnswer is false', () => {
  expect(wrapper).toMatchSnapshot(); 
})

it('should match the snapshot if showAnswer is true', () => {
  wrapper.instance().setState({
    showAnswer: true,
  })
  expect(wrapper).toMatchSnapshot();
})
  
describe('showAnswer', () => {
  it('should set showAnswer to the opposite of its current value', () => {
    expect(wrapper.state('showAnswer')).toEqual(false);
    wrapper.instance().showAnswer();
    expect(wrapper.state('showAnswer')).toEqual(true);
  })
})
})

