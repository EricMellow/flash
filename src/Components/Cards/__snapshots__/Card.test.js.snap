// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot if showAnswer is false 1`] = `ShallowWrapper {}`;

exports[`Card should match the snapshot if showAnswer is true 1`] = `ShallowWrapper {}`;
